src/Examine.Lucene/Providers/LuceneIndex.cs

```diff
@@ -212,11 +234,11 @@ public class LuceneIndex : BaseIndexProvider, IDisposable, IIndexStats, Referenc
         // tracks the latest Generation value of what has been indexed.This can be used to force update a searcher to this generation.
         private long? _latestGen;
 
-        private volatile DirectoryTaxonomyWriter _taxonomyWriter;
-        private ControlledRealTimeReopenThread<SearcherTaxonomyManager.SearcherAndTaxonomy> _taxonomyNrtReopenThread;
+        private volatile DirectoryTaxonomyWriter? _taxonomyWriter;
+        private ControlledRealTimeReopenThread<SearcherTaxonomyManager.SearcherAndTaxonomy>? _taxonomyNrtReopenThread;
 
-        private readonly Lazy<LuceneTaxonomySearcher> _taxonomySearcher;
-        private readonly Lazy<Directory> _taxonomyDirectory;
+        private readonly Lazy<LuceneTaxonomySearcher>? _taxonomySearcher;
```

Comment:
Can this be null? Looks like we are always assigning in the ctor?

Reply:
@nzdev I might have added this ? When the searcher wasn't being created in the protected constuctor. So we can properly remove this nullability again

Reply:
Not nullable, removed ?